### PR TITLE
Add chat streaming support

### PR DESCRIPTION
**Pull request by KaQuMiQ**

### CODE REVIEW

File: src/draive/conversation/call.py [hunk 1]
```diff
@@ -7,17 +10,54 @@
...
+) -> AsyncIterator[ConversationMessage] | ConversationMessage:
     conversation: Conversation = ctx.state(Conversation)
-    return await conversation.completion(
-        instruction=instruction,
-        input=input,
-        memory=memory or conversation.memory,
-        toolset=toolset or conversation.toolset,
-    )
+
+    if stream:
```
Comment (piotr-rarus) [inline src/draive/conversation/call.py:47]: This `if` statement looks weird.

```py
return await conversation.completion(
            instruction=instruction,
            input=input,
            memory=memory or conversation.memory,
            toolset=toolset or conversation.toolset,
            stream=stream,
        )
```

File: src/draive/openai/chat.py [hunk 2]
```diff
@@ -201,3 +363,54 @@ async def _chat_tool_call(
...
+                    else:
+                        tool_call.function = call.function
+                else:
+                    pass
+
+                if call.function.arguments:
+                    if tool_call.function.arguments is not None:
+                        tool_call.function.arguments += call.function.arguments
+                    else:
+                        tool_call.function = call.function
```
Comment (piotr-rarus) [inline src/draive/openai/chat.py:408]: What is the case when interpreter hits this line? We already handled empty and None values.

Comment (KaQuMiQ) [reply src/draive/openai/chat.py]: copy/paste error - it should set the name not the whole function - when function calling is requested names of functions may not be in place yet - fixed

File: src/draive/conversation/call.py (hunk 1 above)
Comment (KaQuMiQ) [reply src/draive/conversation/call.py]: yes it is, I have added a comment - pyright does not handle types correctly if not using literal True/False when calling this

**Review by Jeeejeets — APPROVED** [review]

**Pull request merged**